.github/workflows/tests.yml

```diff
@@ -35,28 +45,46 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
+        
+      - name: Checkout code
+        if: ${{ inputs.repository }} != ${{ github.repository }}
+        uses: actions/checkout@v3
+        with: 
+            repository: UBCSailbot/sailbot_workspace
```

Comment:
Example of what I think this should look like for `network_system` any time it occurs:

```yml
      - name: Checkout code
        if: network_systems != sailbot_workspace
        uses: actions/checkout@v3
        with: 
            repository: UBCSailbot/network_systems
            path: src/network_systems
```

I think you can accomplish this by setting `${{ inputs.repository }}` to `${{ github.event.repository.name }}` in the caller workflow and replace every occurence of `network_systems` in the code block above with `${{ inputs.repository }}`, but I may be wrong

Reply:
You can also change the name of this step to be more descriptive, something like `Checkout ROS package`

Reply:
I'm not too sure about this change, because 
```
      with: 
          repository: UBCSailbot/sailbot_workspace
 ```
 Is the repository it checks out. If its:
```
        with: 
            repository: UBCSailbot/network_systems
```
We aren't checking out sailbot_workspace actions folder into our workflows folder, but instead checking out network_systems which has no action files with our changes.

Reply:
Re-named to ```Checkout ROS package```

Reply:
If the CI is running in sailbot_workspace, only sailbot_workspaces needs to be checked out. If the CI is running in network_systems, sailbot_workspace will need to be checked out **as well as** network_systems in sailbot_workspace's `src/` directory. This will be needed for `colcon-test`, `ament-lint`, and `clang-tidy`.

```yml
      - name: Checkout workspace
        uses: actions/checkout@v3
        repository: UBCSailbot/sailbot_workspace

      - name: Checkout code
        if: ${{ inputs.repository }} != ${{ github.event.repository.name }}
        uses: actions/checkout@v3
        with: 
            repository: UBCSailbot/${{ inputs.repository }}
            path: src/${{ inputs.repository }}
```

The markdown jobs don't use the actions in sailbot_workspace, so they can be checked out as normal.

```yml
      - name: Checkout code
        uses: actions/checkout@v3
        with: 
            repository: UBCSailbot/${{ inputs.repository }}
```

Reply:
I think you mean:
```yml

        with: 
            repository: UBCSailbot/sailbot_workspace

            ```

Otherwise it doesn't run

Reply:
> Otherwise it doesn't run

Is this still the case?

Reply:
Yes, seems to not be able to run:
```yml
      - name: Checkout workspace
        uses: actions/checkout@v3
        repository: UBCSailbot/sailbot_workspace
```
This fixes that:
```yml
      - name: Checkout workspace
        uses: actions/checkout@v3
        with: 
            repository: UBCSailbot/sailbot_workspace
```